src/components/WalletStatementModal/index.native.js

```diff
@@ -1,79 +1,76 @@
-import React from 'react';
+import React, {useCallback, useRef} from 'react';
 import {WebView} from 'react-native-webview';
 import lodashGet from 'lodash/get';
 import {withOnyx} from 'react-native-onyx';
 import _ from 'underscore';
-import withLocalize from '../withLocalize';
+import {useNavigation} from '@react-navigation/native';
 import ONYXKEYS from '../../ONYXKEYS';
-import compose from '../../libs/compose';
 import {walletStatementPropTypes, walletStatementDefaultProps} from './WalletStatementModalPropTypes';
 import FullScreenLoadingIndicator from '../FullscreenLoadingIndicator';
 import * as Report from '../../libs/actions/Report';
-import Navigation from '../../libs/Navigation/Navigation';
 import ROUTES from '../../ROUTES';
 
-class WalletStatementModal extends React.Component {
-    constructor(props) {
-        super(props);
+const IOU_ROUTES = [ROUTES.IOU_REQUEST, ROUTES.IOU_SEND];
+const renderLoading = () => <FullScreenLoadingIndicator />;
 
-        this.authToken = lodashGet(props, 'session.authToken', null);
-        this.navigate = this.navigate.bind(this);
-    }
+function WalletStatementModal({statementPageURL, session}) {
+    const webViewRef = useRef();
+    const authToken = lodashGet(session, 'authToken', null);
+    const {navigate} = useNavigation();
 
     /**
      * Handles in-app navigation for webview links
      *
      * @param {String} params.type
      * @param {String} params.url
      */
-    navigate({type, url}) {
-        if (!this.webview || (type !== 'STATEMENT_NAVIGATE' && type !== 'CONCIERGE_NAVIGATE')) {
-            return;
-        }
+    const handleNavigationStateChange = useCallback(
+        ({type, url}) => {
+            if (!webViewRef.current || (type !== 'STATEMENT_NAVIGATE' && type !== 'CONCIERGE_NAVIGATE')) {
+                return;
+            }
+
+            if (type === 'CONCIERGE_NAVIGATE') {
+                webViewRef.current.stopLoading();
+                Report.navigateToConciergeChat();
+            }
 
-        if (type === 'CONCIERGE_NAVIGATE') {
-            this.webview.stopLoading();
-            Report.navigateToConciergeChat();
-        }
+            if (type === 'STATEMENT_NAVIGATE' && url) {
+                const iouRoute = _.find(IOU_ROUTES, (item) => url.includes(item));
 
-        if (type === 'STATEMENT_NAVIGATE' && url) {
-            const iouRoutes = [ROUTES.IOU_REQUEST, ROUTES.IOU_SEND];
-            const navigateToIOURoute = _.find(iouRoutes, (iouRoute) => url.includes(iouRoute));
-            if (navigateToIOURoute) {
-                this.webview.stopLoading();
-                Navigation.navigate(navigateToIOURoute);
+                if (iouRoute) {
+                    webViewRef.current.stopLoading();
+                    navigate(iouRoute);
```

Comment:
Consider rolling back to `Navigation.navigate`, it runs a lot more logic and checks than the pure `navigate` from `useNavigation`. Changing this could cause unknown side-effects

See here: https://github.com/Expensify/App/blob/b225b53829f8fa4ef5f7a75616469d1fc7d347b9/src/libs/Navigation/Navigation.js#L78-L92